### PR TITLE
Added tests for AdClickthroughWorker

### PR DESCRIPTION
**Pull request by balavor**
@VerizonAdPlatforms/video-partner-sdk-developers: Please review.

### CODE REVIEW

File: AdClickthroughWorkerTests.swift [hunk 1]
```diff
@@ -0,0 +1,112 @@
+//  Copyright 2019, Oath Inc.
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
+
+import XCTest
+@testable import VerizonVideoPartnerSDK
+@testable import PlayerCore
```
Comment (AndriiMoskvin) [inline AdClickthroughWorkerTests.swift]: Do we need `@testable` here?

File: AdClickthroughWorkerTests.swift [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    var isSafariFinishHandlerCalled: Bool!
+    var sut: PlayerViewController.AdClickthroughWorker!
+    
+    override func setUp() {
+        isShowSafariCalled = false
+        isSafariFinishHandlerCalled = false
+        sut = PlayerViewController.AdClickthroughWorker(
+            showSafari: { _, _ in
+                XCTAssertFalse(self.isShowSafariCalled)
+                self.isShowSafariCalled = true
```
Comment (AndriiMoskvin) [inline AdClickthroughWorkerTests.swift]: `Recorder` can be used here.

File: AdClickthroughWorkerTests.swift [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                                         clickthrough: URL(string: "http://")!,
+                                         pixels: AdPixels(),
+                                         id: "")
+    let vpaidCreativeWithoutCllick = AdCreative.VPAID(url: URL(string: "http://")!,
+                                                      adParameters: "",
+                                                      clickthrough: nil,
+                                                      pixels: AdPixels(),
+                                                      id: "")
+    
+    var isShowSafariCalled: Bool!
```
Comment (AndriiMoskvin) [inline AdClickthroughWorkerTests.swift]: Can it be just `false`? I think we don't need `Implicitly Unwrapped Optional` here.

File: AdClickthroughWorkerTests.swift [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
+
+import XCTest
+import PlayerCore
+import SafariServices
+@testable import VerizonVideoPartnerSDK
+
+class AdClickthroughWorkerTests: XCTestCase {
+    let recorder = Recorder()
+    var showSafariHook: (() -> ())!
```
Comment (AndriiMoskvin) [inline AdClickthroughWorkerTests.swift]: I don't think we need to store hooks on level of class - local scope will be enough.

File: AdClickthroughWorkerTests.swift [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+                                                      id: "")
+    
+    var sut: PlayerViewController.AdClickthroughWorker!
+    
+    override func setUp() {
+        let showSafariHook = recorder.hook("showSafariHook")
+        let finishSafariHook = recorder.hook("finishSafariHook")
+        sut = PlayerViewController.AdClickthroughWorker(
+            showSafari: { _, _ in showSafariHook() },
+            safariFinishHandler: { _ in finishSafariHook() })
```
Comment (AndriiMoskvin) [inline AdClickthroughWorkerTests.swift]: Directly pass hook here? - `safariFinishHandler:finishSafariHook`

File: AdClickthroughWorkerTests.swift [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+                                                      pixels: AdPixels(),
+                                                      id: "")
+    
+    var sut: PlayerViewController.AdClickthroughWorker!
+    
+    override func setUp() {
+        let showSafariHook = recorder.hook("showSafariHook")
+        let finishSafariHook = recorder.hook("finishSafariHook")
+        sut = PlayerViewController.AdClickthroughWorker(
+            showSafari: { _, _ in showSafariHook() },
```
Comment (AndriiMoskvin) [inline AdClickthroughWorkerTests.swift]: Same as below - just provide `equatable` block and compare URL's for example.

File: AdClickthroughWorkerTests.swift [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+                                                      pixels: AdPixels(),
+                                                      id: "")
+    
+    var sut: PlayerViewController.AdClickthroughWorker!
+    
+    override func setUp() {
+        let showSafariHook = recorder.hook("showSafariHook")
+        sut = PlayerViewController.AdClickthroughWorker(
+            showSafari: { _, _ in
+                showSafariHook()
```
Comment (AndriiMoskvin) [inline AdClickthroughWorkerTests.swift:39]: Why there is no url comparison?

File: AdClickthroughWorkerTests.swift [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        },
+            safariFinishHandler: recorder.hook("finishSafariHook") { rcd, cmp -> Bool in
```
Comment (AndriiMoskvin) [inline AdClickthroughWorkerTests.swift:41]: `safariFinishHandler: recorder.hook("finishSafariHook")` - this doesn't work?

Comment (balavor) [reply AdClickthroughWorkerTests.swift:41]: It doesn't.

Comment (AndriiMoskvin) [reply AdClickthroughWorkerTests.swift:41]: It works - just checked it.

Comment (balavor) [reply AdClickthroughWorkerTests.swift:41]: Got it. I had compile errors, dunno why =\

**Review by AndriiMoskvin — APPROVED** [review]

**Pull request merged**